src/mlpack/methods/bayesian_ridge/bayesian_ridge.cpp

```diff
@@ -0,0 +1,262 @@
+
+  // Preprocess the data. Center and scale.
+  responsesOffset = CenterScaleData(data,
+                                    responses,
+                                    centerData,
+                                    scaleData,
+                                    phi,
+                                    t,
+                                    dataOffset,
+                                    dataScale);
+
+  // Compute this quantities once and for all.
+  const arma::colvec vecphitT = phi * t.t();
+
+  // Enforce symmetry of the covariance matrix before eig_sym.
+  const arma::mat phiphiT =  arma::symmatu(phi * phi.t());
+
+  if (arma::eig_sym(eigval, eigvec, phiphiT) == false)
+  {
+    Log::Warn << "Eigen Decomposition failed as Eigen Value "
+              << "does not exists ." << std::endl;
+    return -1;
+  }
+
+  // Initialize the hyperparameters and
+  // begin with an infinitely broad prior.
+  alpha = 1e-6;
+  beta =  1 / (var(t) * 0.1);
+
+  unsigned short nIterMax = 50;
+  unsigned short i = 0;
+  double deltaAlpha = 1, deltaBeta = 1, crit = 1;
+  arma::mat matA = arma::eye<arma::mat>(data.n_rows, data.n_rows);
+
+  while ((crit > tol) && (i < nIterMax))
+  {
+    deltaAlpha = -alpha;
+    deltaBeta = -beta;
+
+    // Compute the posterior statistics.
+    // with inv()
+    matA.diag().fill(alpha);
+    // inv is used instead of solve because we need the covariance matrix to
+    // compute the prediction uncertainties. If solve is used, matCovariance
+    // must be comptuted at the end of the loop.
+    matCovariance = inv_sympd(matA + phiphiT * beta);
+    omega = (matCovariance * vecphitT) * beta;
+
+    // // with solve()
+    // matA.diag().fill(alpha/ beta);
+    // omega = solve(matA + phiphiT, vecphitT);
+
+    // Update alpha.
+    eigvali = eigval * beta;
+    gamma = sum(eigvali / (alpha + eigvali));
+    alpha = gamma / dot(omega.t(), omega);
+
+    // Update beta.
+    const arma::rowvec temp = t - omega.t() * phi;
+    beta = (data.n_cols - gamma) / dot(temp, temp);
+
+    // Comptute the stopping criterion.
+    deltaAlpha += alpha;
+    deltaBeta += beta;
+    crit = abs(deltaAlpha / alpha + deltaBeta / beta);
+    i++;
+  }
+  Timer::Stop("bayesian_ridge_regression");
+  return Rmse(data, responses);
+}
+
+void BayesianRidge::Predict(const arma::mat& points,
+                            arma::rowvec& predictions) const
+{
+  // y_hat = w^T * (X - mu) / sigma + y_mean.
+  predictions = omega.t() *
+  ((points.each_col() - dataOffset).each_col() / dataScale) + responsesOffset;
+}
+
+void BayesianRidge::Predict(const arma::mat& points,
+                            arma::rowvec& predictions,
+                            arma::rowvec& std) const
+{
+  // Center and scaleData the points before applying the model.
+  const arma::mat X = (points.each_col() - dataOffset).each_col() / dataScale;
+  predictions = omega.t() * X + responsesOffset;
+  std = sqrt(Variance() + sum((X % (matCovariance * X)), 0));
```

Comment:
In fact it may be quicker to store the Cholesky decomposition of the covariance matrix, and then use that here instead of `matCovariance`.  I'm not 100% sure.

Reply:
Because the triangular are the same size of the covariance I suppose that it it is the same. What do you think ?